### DISCUSSION TITLE
How to run renovate bot locally?

### DISCUSSION THREAD

**Discussion by xuhdev** [renovatebot/renovate#8144] [Request Help] [answered]
Is it possible to run renovate bot locally, and generate patch files which the bot would be suggesting?

**Comment by rarkins** [comment, marked as answer]
No, it's not possible

**Reply by xakraz** [reply]
Sorry for digging up this discussion, but could it be a feature request?

Some of Renovate's config properties are not "easy" to test.
We can only push, let Renovate run, look at the results, and "fix" again.

**Reply by rarkins** [reply]
It can be a feature request (maybe already exists) but requires some specifics about exactly how you think it would work. "run locally" is too vague on its own

**Reply by HonkingGoose** [reply]
We already have a feature request for it:

- https://github.com/renovatebot/renovate/issues/17339

The issue mentions using Docker images to run Renovate locally.
